helper/grpc-middleware/logging/options.go

```diff
@@ -0,0 +1,89 @@
+package logging
+
+import (
+	"github.com/hashicorp/go-hclog"
+	"google.golang.org/grpc/codes"
+)
+
+type options struct {
+	levelFunc CodeToLevel
+}
+
+var defaultOptions = &options{}
+
+type Option func(*options)
+
+func evaluateClientOpt(opts []Option) *options {
+	optCopy := &options{}
+	*optCopy = *defaultOptions
```

Comment:
If the goal here is to work on a copy of the default options, could we make that more explicit with either a `func newDefaultOptions() *options` constructor or a `func (opts *options) Copy()` method?

(I might be missing something though, as we're not passing in any options yet and don't have anything other than the one `levelFunc` option currently, so I might be missing an obvious future bit of code that makes that a bad idea.)

Reply:
this is essentially a constructor - eventually we'll also have options that only apply to servers, that might have some different defaults, and I'm not quite sure how we'll structure it yet. So it's mostly laziness till we figure that out, which will probably be when Nomad adopts gRPC as a net/rpc replacement for some subset of RPCs

Reply:
Sounds good 👍